src/prefect/flows.py

```diff
@@ -148,7 +149,12 @@ def __init__(
             raise_on_invalid_name(name)
 
         self.name = name or fn.__name__.replace("_", "-")
+
+        if flow_run_name is not None:
+            if not isinstance(flow_run_name, str) and not callable(flow_run_name):
+                raise TypeError("'flow_run_name' is neither a string nor a callable")
```

Comment:
Generally we want to give more information to help debug
```suggestion
                raise TypeError(f"Expected string or callable for 'flow_run_name'; got {type(flow_run_name).__name__}.")
```

Reply:
done